listaLigada.h

```diff
@@ -0,0 +1,450 @@
+        Cuenta las apariciones del elemento dado que hay dentro de la lista. Devuelve 0 si no
+        encuentra nada.
+        */
+        int contar(Tipo elem);
+
+        /*
+        Devuelve el dato que esta en el nodo cuyo indice es el pasado por parametro.
+        */
+        Tipo& operator[](int indice);
+
+        /*
+        Recibe otra lista y combina sus elementos de forma que el ultimo elemento de la lista tiene
+        como nodo proximo al primero de 'otraLista'.
+        No muta ninguna de las dos listas y devuelve una copia de la union de estas.
+        
+        Lista<Tipo>*& operator+(Lista<Tipo>* otraLista);
+        */
+
+        /*
+        Compara dos listas y devuelve 'true' si tienen la msima cantidad de elementos y estos son los
+        mismos. Si no devuelve 'false'.
+        
+        bool operator==(Lista<Tipo>* otraLista);
+        */
+
+        /*
+        Imprime por pantalla todos los elementos de la lista, para que se los pueda ver comodamente.
+        */
+        void imprimir();
+};
+
+/* ----- DEFINICIONES ----- */
+
+template <class Tipo>
+Lista<Tipo>::Lista() {
+
+    prim = NULL;
+    cant = 0;
+}
+
+// template <class Tipo>
+// Lista<Tipo>::Lista(Lista<Tipo>* otraLista) {
+
+//     Nodo<Tipo>* aux1;
+//     Nodo<Tipo>* aux2 = otraLista.prim;
+
+//     while (aux2 && aux2->prox()) {
+
+//         aux1 = new Nodo<Tipo>(aux2->dato(), aux2->prox());
+
+//         aux1 = aux1->prox();
+//         aux2 = aux2->prox();
+//     }
+// }
+
+template <class Tipo>
+Lista<Tipo>::~Lista() {
+
+    if (prim) {
+
+        Nodo<Tipo>* aux = prim;
+        while (aux) {
+
+            prim = prim->prox();
+            delete aux;
+            aux = prim;
+        }
+    }
+}
+
+template <class Tipo>
+bool Lista<Tipo>::estaVacia() {
+
+    return prim == NULL;
+}
+
+template <class Tipo>
+int Lista<Tipo>::len() {
+
+    return cant;
+}
+
+template <class Tipo>
+void Lista<Tipo>::agregarPrin(Tipo elem) {
+
+    prim = new Nodo<Tipo>(elem, prim);
+    cant++;
+}
+
+template <class Tipo>
+void Lista<Tipo>::agregarFin(Tipo elem) {
+
+    Nodo<Tipo>* aux = prim;
+    while (aux && aux->prox()) {
+
+        aux = aux->prox();
+    }
+
+    if (aux) {
+
+        aux->setProx(new Nodo<Tipo>(elem, aux->prox()));
+
+    } else {
+
+        prim = new Nodo<Tipo>(elem, prim);
+    }
+
+    cant++;
+}
+
+template <class Tipo>
+void Lista<Tipo>::insertar(int indice, Tipo elem) {
+
+    if (indice < 0 || indice > cant) {
+        throw "El indice indicado esta fuera de rango";
+    }
+
+    if (indice == 0) {
+
+        agregarPrin(elem);
+
+    } else {
+
+        Nodo<Tipo>* posterior = prim;
+        for (int contador = 0; (posterior) && (posterior->prox()) && (contador < indice - 1); contador++) {
+
+            posterior = posterior->prox();
+        }
+
+        if (posterior) {
+
+            posterior->setProx(new Nodo<Tipo>(elem, posterior->prox()));
+            cant++;
+
+        } else {
+
+            agregarFin();
```

Comment:
El que escribio esto sin argumento debe ser muy noob

```suggestion
            agregarFin(elem);
```